### PR TITLE
Query: Convert enum to underlying CLR type in type mapping when type …

### LINKED ISSUE #21770: InvalidCastException when cast enum to underlying type in linq and compare to an different integer type in db
Not sure if it's an issue an efcore side or provider side, but see similar Exception in `Pomelo.EntityFrameworkCore.MySql` and `Npgsql.EntityFrameworkCore.PostgreSQL`
https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/issues/1131
>System.InvalidCastException: 'Unable to cast object of type 'ZZZ' to type 'System.Int32'.'

https://github.com/npgsql/efcore.pg/issues/1436
>System.InvalidCastException:“Can't write CLR type ZZZ with handler type Int32Handler”

### Steps to reproduce

``` C#
public enum ZZZ : byte { AAA = 1 }

var a = ZZZ.AAA;
ctx.AnyTables.Where(d => d.AnyInt32Field == (byte)a).FirstOrDefault();
```

An workaroud is do the cast out of linq, 
```cs
var a = (byte)ZZZ.AAA;
ctx.AnyTables.Where(d => d.AnyInt32Field == (byte)a).FirstOrDefault();
```
or cast to object first inside linq 
```cs
var a = ZZZ.AAA;
ctx.AnyTables.Where(d => d.AnyInt32Field == (byte)(object)a).FirstOrDefault();
```

### Further technical details

EF Core version: 3.1.6/5.0.0preview7
Database provider: `Pomelo.EntityFrameworkCore.MySql 3.1.2`/`Npgsql.EntityFrameworkCore.PostgreSQL 3.1.4/5.0.0preview7`
Target framework: .NET Core 3.1/5.0preview7
Operating system: win10-x64
IDE: (e.g. Visual Studio 2019 16.3)

### PR DESCRIPTION
**Pull request by smitpatel**
…mapping is not for enum

Resolves #21770

### CODE REVIEW

**Comment by smitpatel** [comment]
@lauxjpn @roji - Added tests pass without fix for all built-in providers. It would be useful if you can verify the fix on your end too.

**Comment by roji** [comment]
@smitpatel tests pass for me on Npgsql :+1:

File: src/EFCore.Relational/Storage/RelationalTypeMapping.cs [hunk 1]
```diff
@@ -485,6 +485,15 @@ public virtual bool IsFixedLength
             parameter.ParameterName = name;
 
             value = ConvertUnderlyingEnumValueToEnum(value);
+            // We preserve convert nodes around enum to get actual enum value for type mapping
```
Comment (roji) [inline src/EFCore.Relational/Storage/RelationalTypeMapping.cs:488]:
```suggestion

            // We preserve convert nodes around enum to get actual enum value for type mapping
```

**Review by roji — APPROVED** [review]

File: src/EFCore.Relational/Storage/RelationalTypeMapping.cs (hunk 1 above)
Comment (smitpatel) [reply src/EFCore.Relational/Storage/RelationalTypeMapping.cs:488]: Just added a new line before comment?

Comment (roji) [reply src/EFCore.Relational/Storage/RelationalTypeMapping.cs:488]: Yep :)

**Review by AndriySvyryd — APPROVED** [review]

**Comment by clement128** [comment]
Can this also merge to efcore 5.0? thanks

**Pull request merged**